### PR TITLE
allow using custome configuration directory for psalm

### PR DESCRIPTION
**Pull request by azjezz**
I was trying to use this project with PSL, and it didn't work well, since PSL stores all configuration files in `config/` directory.

using this patch, I'm able to keep the same structure, by setting `PROJECT_CONFIG_DIRECTORY` env variable to `config`.

### CODE REVIEW

**Comment by azjezz** [comment]
and now it works after 20 minutes of debugging 😎 
image

File: bin/roave-infection-static-analysis-plugin [hunk 1]
```diff
@@ -59,7 +60,10 @@ use function var_export;
     $makeAnalyzer = static function () use ($projectPath): ProjectAnalyzer {
         RuntimeCaches::clearAll();
 
-        $config = Config::getConfigForPath($projectPath, $projectPath);
+        $configDirectory = getenv('PROJECT_CONFIG_DIRECTORY') ?: '';
+        $configPath = '' !== $configDirectory ? ($projectPath . '/' . $configDirectory) : $projectPath;
```
Comment (Ocramius) [inline bin/roave-infection-static-analysis-plugin:64]: This is what the `-c` flag does for psalm, no? Is it not being respected at all?

The problem here is that we obviously have flags from both infection and psalm.

IMO, preserving the `psalm.xml` or `psalm.xml.dist` at the root of a software project is simpler than introducing more customization to this project to make ends meet elsewhere.

**Review by Ocramius — CHANGES_REQUESTED** [review]

**Comment by Ocramius** [comment]
Potential solution that I would prefer (although unsure if I can push this much responsibility down the line): add `--psalm-<flag-or-parameter-name>` capabilities to our CLI :thinking: 

Please don't implement any of it - don't want to waste anyone's time until we got a bit of an agreement /cc @WyriHaximus

**Comment by azjezz** [comment]
> Potential solution that I would prefer (although unsure if I can push this much responsibility down the line): add --psalm-<flag-or-parameter-name> capabilities to our CLI

I don't think that is actually worth doing, if we can make this part lazy ( maybe move fetching the service part to the callback? ):

https://github.com/Roave/infection-static-analysis-plugin/blob/57e194e643de67b8f38299f13fca88b263f21a4d/src/Roave/InfectionStaticAnalysis/Bootstrapper.php#L21-L28

This would mean that all of infection configurations will be respected ( which solves #27 ), then we can add 1 flag that points to psalm configuration ( --psalm-config="..." ), and remove that flag from `$argv` before running infection to avoid Symfony Console complaining about unknown flags/options.

The reason i think we should only support -c psalm flag, is that all other flags don't make sense in this context.

output flags -> we don't show output
baseline flags -> doesn't make sense to update/create baseline during tests
surfacing issues flags -> doesn't make sense to be used in tests
plugins flags -> plugins can be added to psalm.xml instead
reports flags -> we don't show reports/output
caching flags -> user is not really in charge of caching in this case.
thread flag -> use infection threads instead.
memory limit flag -> same as `php -dmemory_limit=X vendor/bin/roave-infection-static-analysis-plugin`

**Comment by Ocramius** [comment]
Dunno, I often run `vendor/bin/psalm --no-cache` because it is extremely unreliable **with** the cache, hence why I wanted to ask.

Supporting only `-c` could work as well, if it doesn't conflict with a `-c` from infection

**Pull request closed without merging**